### ISSUE TITLE
Provide a general mechanism for adding extra logic when an ObservableProperty changes

### ISSUE THREAD

**Issue by mikegoatly** [CommunityToolkit/dotnet#171]
### Overview

It's sometimes useful to invoke a method specifically for a property when it changes value, e.g.:

``` csharp
[ObservableProperty]
[AlsoInvokeOnChanged(nameof(OnMyPropertyChanged))]
private bool myProperty;

private void OnMyPropertyChanged(bool newValue)
{
    // Do something here
}
```

This would allow for people to manage edge cases that the auto-generated code can't currently provide (it would have been a workaround for #125)

### API breakdown

``` csharp
[ObservableProperty]
// The AlsoInvokeOnChanged attribute is associated to the ObservableProperty, specifying the name of the method
// that should be executed.
[AlsoInvokeOnChanged(nameof(OnMyPropertyChanged))]
private bool myProperty;

// A single parameter is required, and will be passed the new value that has been set: 
private void OnMyPropertyChanged(bool newValue)
{
    // ...
}
```

### Usage example

``` csharp
public partial class OptionsExampleViewModel : ObservableObject
{
    [ObservableProperty(nameof(OnRenderingShowTextLayoutBoundsChanged))]
    [AlsoInvokeOnChanged(nameof(OnToggleOneChanged))]
    private bool toggleOne;

    [ObservableProperty(nameof(OnRenderingShowTextLayoutBoundsChanged))]
    private bool toggleTwo;

    private void OnToggleOneChanged(bool newValue)
    {
        if (newValue == false)
        {
            // When toggling the first option off, also ensure the second is turned off.
            ToggleTwo = false;
        }
    }
}
```

### Breaking change?

No

### Alternatives

Either:

1) Avoid using the `ObservableProperty` for the given property and implement the get and set manually
2) Override `OnPropertyChanged` and check the name of the property that's just changed and handle the logic there. This is a bit ugly and requires (marginally) more processor time with the required string comparisons and casting of the changed value back to the required type. 

### Help us help you

Yes, but only if others can assist

**Comment by Sergio0694** [comment]
Hey there! This is actually already available in 8.0.0-Preview2 😄
You can see more info [here](https://github.com/CommunityToolkit/dotnet/issues/157#issuecomment-1069027771).

**Comment by mikegoatly** [comment]
@Sergio0694 Awesome, thanks. I hadn't thought about checking for an implicitly named property changed method. I should have looked at the generated code; lesson learned!

I had started building my own code generators for doing this sort of thing, but these are much more fully featured, and I'm very happy to not have to write all the code myself going forwards!
